mcf-browser/src/back-end/data.ts

```diff
@@ -0,0 +1,250 @@
+    this.observationAbout = cleanVariable(observationAbout);
+    this.provenance = cleanVariable(provenance);
+    this.measurementMethod = cleanVariable(measurementMethod);
+    this.observationPeriod = cleanVariable(observationPeriod);
+    this.unit = cleanVariable(unit);
+    this.scalingFactor = scalingFactor;
+
+    this.id = Series.toID(
+        variableMeasured,
+        observationAbout,
+        provenance,
+        measurementMethod,
+        observationPeriod,
+        unit,
+        scalingFactor,
+    );
+  }
+
+  /**
+   * Generates a unique ID for a series given the values of
+   * the properties of a Series object
+   * @param {string} variableMeasured the variableMeasured of a series
+   * @param {string} observationAbout the observationAbout of a series
+   * @param {string} provenance the provenance of a series
+   * @param {string} measurementMethod the measurementMethod of a series
+   * @param {string} observationPeriod the observationPeriod of a series
+   * @param {string} unit the unit of a series
+   * @param {number} scalingFactor the scalingFactor of a series
+   * @return {string} the id
+   */
+  static toID(
+      variableMeasured?: string,
+      observationAbout?: string,
+      provenance?: string,
+      measurementMethod?: string,
+      observationPeriod?: string,
+      unit?: string,
+      scalingFactor?: number,
+  ) : string {
+    const facetList = [
+      variableMeasured ? variableMeasured : '',
+      observationAbout ? observationAbout : '',
+      provenance ? provenance : '',
+      measurementMethod ? measurementMethod : '',
+      observationPeriod ? observationPeriod : '',
+      unit ? unit : '',
+      scalingFactor ? scalingFactor.toString() : '',
+    ];
+
+    return facetList.join(ID_DELIMITER);
+  }
+
+  /** Takes in an ID string and returns the corresponding values in an object
+   * @param {string} id the id of a Series generated by Series.toID
+   * @return {Object<string, string | number>} the properties of a
+   *  series parsed from id
+   */
+  static fromID(id: string) : SeriesIdObject {
+    const [
+      variableMeasured,
+      observationAbout,
+      provenance,
+      measurementMethod,
+      observationPeriod,
+      unit,
+      scalingFactor,
+    ] = id.split(ID_DELIMITER);
+
+    const parseString = (str: string) => (str === '') ? undefined : str;
+
+    return {
+      variableMeasured: parseString(variableMeasured),
+      observationAbout: parseString(observationAbout),
+      provenance: parseString(provenance),
+      measurementMethod: parseString(measurementMethod),
+      observationPeriod: parseString(observationPeriod),
+      unit: parseString(unit),
+      scalingFactor: parseFloat(scalingFactor),
+    };
+  }
+
+  /** Sorts the datapoints and rewrites this.data
+   * to the sorted version
+   */
+  sortData() {
+    const data = this.data.map((dataPoint) => {
+      return {
+        ...dataPoint,
+        xNum: Date.parse(dataPoint.x),
```

Comment:
Add a clarifying comment what this is for?

Also, if the dates are well-formed ISO 8601, I suspect string comparison should work.

Reply:
Oh that's cool I didn't know ISO 8601 was string comparable, it's changed in the next commit!